boards/Kconfig

```diff
@@ -1584,6 +1584,13 @@ config ARCH_BOARD_BEAGLEBONE_BLACK
 	---help---
 		This is the port of NuttX to the TI Beaglebone Black board.
 
+config ARCH_BOARD_T3_GEM_O1
+	bool "T3 Gemstone O1"
+	depends on ARCH_CHIP_AM67
+	---help---
+		This is the port of NuttX to the T3 Gemstone O1 board.
+		See https://docs.t3gemstone.org/en/boards/o1/introduction for further information.
```

Comment:
Someone else can weigh in, but Kconfig should only link to NuttX docs imo. If you include this link on the NuttX doc page, that's fine, but all the necessary information for how to use the board port should exist on the NuttX docs and not externally.